src/task.rs

```diff
@@ -2342,3 +2360,76 @@ impl<'a> Runner<'a> {
         Ok(all_ok)
     }
 }
+
+fn resolve_path(target: &String) -> String {
+    path_from(current_dir().unwrap(), target.as_str()).to_str().unwrap().to_string()
+}
+/// https://stackoverflow.com/questions/68231306/stdfscanonicalize-for-files-that-dont-exist
+/// build a usable path from a user input which may be absolute
+/// (if it starts with / or ~) or relative to the supplied base_dir.
+/// (we might want to try detect windows drives in the future, too)
+pub fn path_from<P: AsRef<Path>>(
+    base_dir: P,
+    input: &str,
+) -> PathBuf {
+    let tilde = Regex::new(r"^~(/|$)").unwrap();
+    if input.starts_with('/') {
+        // if the input starts with a `/`, we use it as is
+        input.into()
+    } else if tilde.is_match(input) {
+        // if the input starts with `~` as first token, we replace
+        // this `~` with the user home directory
+        PathBuf::from(
+            &*tilde
+                .replace(input, |c: &Captures| {
+                    if let Some(user_dirs) = UserDirs::new() {
+                        format!(
+                            "{}{}",
+                            user_dirs.home_dir().to_string_lossy(),
+                            &c[1],
+                        )
+                    } else {
+                        // warn!("no user dirs found, no expansion of ~");
+                        c[0].to_string()
+                    }
+                })
+        )
+    } else {
+        // we put the input behind the source (the selected directory
+        // or its parent) and we normalize so that the user can type
+        // paths with `../`
+        normalize_path(base_dir.as_ref().join(input))
+    }
+}
+
+
+/// Improve the path to try remove and solve .. token.
+///
+/// This assumes that `a/b/../c` is `a/c` which might be different from
+/// what the OS would have chosen when b is a link. This is OK
+/// for broot verb arguments but can't be generally used elsewhere
+///
+/// This function ensures a given path ending with '/' still
+/// ends with '/' after normalization.
+pub fn normalize_path<P: AsRef<Path>>(path: P) -> PathBuf {
+    let ends_with_slash = path.as_ref()
+        .to_str()
+        .map_or(false, |s| s.ends_with('/'));
+    let mut normalized = PathBuf::new();
+    for component in path.as_ref().components() {
+        match &component {
+            Component::ParentDir => {
+                if !normalized.pop() {
+                    normalized.push(component);
+                }
+            }
+            _ => {
+                normalized.push(component);
+            }
+        }
+    }
+    if ends_with_slash {
+        normalized.push("");
+    }
+    normalized
+}
```

Comment:
Is it possible to directly reference a library for this stuff or do you think we need to have this logic in the source?

Reply:
I couldn't find a proper library for these, so If you think there's a one, please let me know! Though I think these are small and minimal functions.